app/controllers/application_controller.rb

```diff
@@ -12,6 +13,17 @@ def authenticate_admin!
     end
   end
 
+  ##
+  # Forbid the request if the back_uri parameter is being set to create
+  # offsite links.  The parameter is used in various views throughout the
+  # application.
+  def sanitize_back_uri!
+    if params[:back_uri].present?
+      ok = request.protocol + request.host
+      render nothing: true, status: 403 if !params[:back_uri].start_with? ok
```

Comment:
FWIW, the other thing I was considering here was whether the `:back_uri` handling should happen at the level of the [view](https://github.com/dpla/frontend/blob/develop/app/views/shared/_search_panel.html.haml#L3-L4), and whether the parameter would just be ignored if it was somehow invalid. This is probably a cleaner solution FWIW.